src/main/java/fi/aalto/cs/apluscourses/presentation/MainViewModelUpdater.java

```diff
@@ -135,25 +131,31 @@ Course getCourse(@NotNull URL courseUrl) {
     return updatableModules;
   }
 
-  private void runUpdateLoop() throws InterruptedException {
-    while (true) {
-      URL courseUrl = getCourseUrl();
-      Course course = getCourse(courseUrl);
-      // If parsing the course configuration file fails, then we just silently go back to sleep.
-      // For an example, if the internet connection was down, then we may succeed when we try
-      // again after sleeping.
-      if (course != null) {
-        // TODO: what do we actually do with the list of updatable modules? Notify the user?
-        List<Module> updatableModules = getUpdatableModules(course);
-        // The project may have closed while we parsed the course configuration file and computed
-        // the updatable modules, in which case we throw the result away and stop this updater.
-        ReadAction.run(this::interruptIfProjectClosed);
-        mainViewModelProvider
-            .getMainViewModel(aplusProject.getProject())
-            .courseViewModel
-            .set(new CourseViewModel(course));
-      }
-      Thread.sleep(updateInterval); // Good night :)
+  private void updateMainViewModel(@NotNull Course newCourse) {
+    CourseViewModel courseViewModel = mainViewModel.courseViewModel.get();
+    if (courseViewModel == null) {
+      mainViewModel.courseViewModel.set(new CourseViewModel(newCourse));
+      return;
+    }
+
+    Course current = courseViewModel.getModel();
+    /*
+     * Updating the course view model while modules are installing leads to "Error in dependencies"
+     * in the modules tool window. The installations actually still work, and the error state
+     * disappears on the next update, but it's still bad UI/UX. Therefore we check if the currently
+     * loaded course has any "active" components, and only update the course view model if it
+     * doesn't.
+     */
+    boolean hasActiveComponents = current.getComponents()
+        .stream()
+        .anyMatch(component -> {
+          int state = component.stateMonitor.get();
+          int depState = component.dependencyStateMonitor.get();
+          return state == Component.FETCHING || state == Component.LOADING
+              || depState == Component.DEP_WAITING;
```

Comment:
Nit: This could be a method, e.g. `Component::isActive`, so it could be unit tested.